### ISSUE TITLE
Mouse component should store the mouse events history and rely on it to detect responses

### ISSUE THREAD

**Issue by lucaf** [psychopy/psychopy#1283]
Psychopy (module event.py) does store only the last mouse event occurred. This happens asynchronously respect to any routine loop where normally mouse component is checked, and might cause the unlikely but possible miss of events. I propose to improve the mouse component so that it would take in account all the mouse events occurring in between one call to getPressed() and the successive.

**Comment by hoechenberger** [comment]
Hi @lucaf, you [mentioned ](http://discourse.psychopy.org/t/windows7-use-of-touchscreen-monitor-as-mouse-unexpected-behavior/1317/5?u=richard)on the forum that you already have implemented this behavior. Maybe you can file a PR so others can review this approach an evaluate it for possibility of inclusion? Thanks!

**Comment by lucaf** [comment]
Yep. Well, that needs to be revised. It was a quick patch: I wrote a new class  and I used it in place of the Mouse component. Let me look at it again when I have a bit of time left, so I can come out with something which integrates better.

**Comment by peircej** [comment]
I imagine this doesn't get around the issue of polling for events at more than the monitor frame rate (so it's less complete than the iohub solution?)

**Comment by lucaf** [comment]
Unfortunately not. Nevertheless, it ensures that no events are lost.

**Comment by isolver** [comment]
@peircej, can we close this?
